### ISSUE TITLE
Xcode 9 Beta Warnings

### ISSUE THREAD

**Issue by EverGod** [InderKumarRathore/DeviceGuru#43]
Xcode compiler complains "Overlapping accesses to 'name', but modification requires exclusive access; consider copying to a local variable"

DeviceGuru.swift file (lines 23 & 25).
![warnings](https://user-images.githubusercontent.com/518485/28664727-0c26d270-72ca-11e7-8679-af83fd0e743f.png)

23. sysctl(&name, 2, nil, &size, &name, 0)
25. sysctl(&name, 2, &hw_machine, &size, &name, 0)

**Comment by DarkoDamjanovic** [comment]
Just created a second "name" var. Like this:

```
  class open func hardwareString() -> String {
    var name: [Int32] = [CTL_HW, HW_MACHINE]
    var name2: [Int32] = [CTL_HW, HW_MACHINE]
    var size: Int = 2
    sysctl(&name, 2, nil, &size, &name2, 0)
    var hw_machine = [CChar](repeating: 0, count: Int(size))
    sysctl(&name, 2, &hw_machine, &size, &name2, 0)
    
    let hardware: String = String(cString: hw_machine)
    return hardware
  }
```

Works for me.

**Comment by phatmann** [comment]
@DarkoDamjanovic Can you submit a pull request?

**Comment by DarkoDamjanovic** [comment]
@phatmann I am not sure if this fix is the correct way of doing it. Do you know more?

**Comment by phatmann** [comment]
Seems like a perfect fix to me.

**Comment by DarkoDamjanovic** [comment]
I just saw that there is already a pull request for this error. And it seems better then mine:

https://github.com/InderKumarRathore/DeviceGuru/pull/41

**Comment by InderKumarRathore** [comment]
Hi guys sorry for the delay, but I was waiting for the GM Seed. I've downloaded the latest GM Seed xcode9 and merged the pull request #41
